gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -138,6 +142,17 @@ impl ClnRpcService {
                     handle.await?
                 },
             )
+            .subscribe(
+                "shutdown",
+                |plugin: Plugin<Arc<ClnHtlcInterceptor>>, _: serde_json::Value| async move {
+                    info!("Shutting down...");
```

Comment:
Can we say something like `received "shutdown" event from lightningd ... shutting down`?

Reply:
Might also be nice to make a comment about when this event fires. E.g. it doesn't run if `lightningd` is killed?

Reply:
I agree.  technically it's a notification. `Received shutdown notification from lightningd`

Might be worth [to link to the shutdown notification documentation](https://lightning.readthedocs.io/PLUGINS.html?highlight=shutdown#shutdown) which is rather short but concise.

> Send in two situations: lightningd is (almost completely) shutdown, or the plugin stop command has been called for this plugin. In both cases the plugin has 30 seconds to exit itself, otherwise it’s killed.
> 
> In the shutdown case, plugins should not interact with lightnind except via (id-less) logging or notifications. New rpc calls will fail with error code -5 and (plugin’s) responses will be ignored. Because lightningd can crash or be killed, a plugin cannot rely on the shutdown notification always been send.